### DISCUSSION TITLE
Can I use the existing Z-axis switch with a 3DTouch?

### DISCUSSION THREAD

**Discussion by bitbarista** [Jyers/Marlin#1481] [Q&A]
Hi, I have installed a 3DTouch on my Voxelab Aquila and occasionally the nozzle will crash into the bed during homing. It's very random but concerning when it does,  so I would like to re-instate the old Z-axis switch but use the 3DTouch for levelling. Is this possible whilst still using the BLTouch-5x5-G32.bin firmware and if so then how would I do this?

**Comment by FinlessBob** [comment]
Odd, I have never had a bed crash using a BLTouch.

**Reply by adam3654** [reply]
Early on, I did have a couple bed crashes, but I am pretty certain it was some kind of user error. I never narrowed down why it happened but as I got cura settings dialed in, and better understood flashing firmware with resetting of the eeprom, it has never happened again. I would suggest double checking your slicer settings (doing things in the correct order), and make sure you reset eeprom after flashing firmware ( pretty sure M502, M500, M501 is the correct sequence).

**Comment by bitbarista** [comment]
@adam3654 it's too random to be a firmware issue. I think I may have narrowed it down to being electrical interference, which seems to be prevalent when the hot end heater is on, so I suspect it is the PWM interfering with the 3DTouch. I've now changed the extension cable to a screened one and the 3DTouch cable is now twisted groups. A little early to say for sure, but so far no crashes.

**Reply by bitbarista** [reply]
One of the other characteristics I had noticed was that when creating a new bed levelling mesh it would sometimes not complete all probing points. Since changing to screened and twisted cables I have performed a few and none have failed.

**Comment by bitbarista** [comment]
Just to update all who read this, the nozzle crashing issue was almost definitely related to the 3DTouch picking up interference. Since changing the extension cable to screened cable and twisting the 3dTouch cables, I have not had a single crash after many homings and levellings.
